src/plugins/intel_gpu/src/graph/impls/onednn/utils.cpp

```diff
@@ -297,150 +333,264 @@ static const std::map<cldnn::format::type, dnnl::memory::format_tag> format_map_
+                break;
+
+            case dnnl::memory::format_tag::abced:
+                strides.push_back(1);
+                strides.push_back(padded_dims[0]);
+                strides.push_back(padded_dims[0] * padded_dims[1]);
+                strides.push_back(padded_dims[0] * padded_dims[1] * padded_dims[2]);
+                strides.push_back(padded_dims[0] * padded_dims[1] * padded_dims[2] * padded_dims[4]);
+                break;
+
+            case dnnl::memory::format_tag::abcdfe:
+                strides.push_back(1);
+                strides.push_back(padded_dims[0]);
+                strides.push_back(padded_dims[0] * padded_dims[1]);
+                strides.push_back(padded_dims[0] * padded_dims[1] * padded_dims[2]);
+                strides.push_back(padded_dims[0] * padded_dims[1] * padded_dims[2] * padded_dims[3]);
+                strides.push_back(padded_dims[0] * padded_dims[1] * padded_dims[2] * padded_dims[3] * padded_dims[5]);
+                break;
+
+            case dnnl::memory::format_tag::ba:
+                strides.push_back(1);
+                strides.push_back(padded_dims[1]);
+                break;
+
+            default: {
+                auto pitches = _layout.get_pitches();
+                strides.assign(pitches.begin(), pitches.end());
+                break;
+            }
         }
-        dnnl::memory::desc res(dims, dt, strides);
-        return res;
-    } else {
-        dnnl::memory::format_tag fmt = target_fmt == dnnl::memory::format_tag::undef ? convert_data_format(l.format) : target_fmt;
-        dnnl::memory::desc res(dims, dt, fmt);
-        return res;
+
+        return strides;
     }
+};
+
+// Public API functions using the builder
+dnnl::memory::desc layout_to_memory_desc(const cldnn::layout& l, dnnl::memory::format_tag target_fmt) {
+    return MemoryDescriptorBuilder(l, target_fmt).build();
+}
+
+dnnl::memory::desc layout_to_memory_desc_flatten(const cldnn::layout& l, dnnl::memory::format_tag target_fmt) {
+    return MemoryDescriptorBuilder(l, target_fmt).as_flattened().build();
+}
+
+dnnl::memory::desc layout_to_memory_desc_blocked(const cldnn::layout& l, dnnl::memory::format_tag target_fmt) {
+    return MemoryDescriptorBuilder(l, target_fmt).keep_rank().build();
+}
+
+dnnl::memory::desc layout_to_memory_desc_grouped(const cldnn::layout& l, dnnl::memory::format_tag target_fmt) {
+    return MemoryDescriptorBuilder(l, target_fmt).keep_rank().build();
 }
+
+dnnl::memory::desc layout_to_memory_desc_strides(const cldnn::layout& l, dnnl::memory::format_tag target_fmt) {
+    return MemoryDescriptorBuilder(l, target_fmt).with_strides().build();
+}
+
+dnnl::memory::desc layout_to_memory_desc(cldnn::layout l, bool use_default_format, bool is_output_blocked) {
+    auto md = MemoryDescriptorBuilder(l, dnnl::memory::format_tag::undef);
+
+    if (use_default_format) {
+        md.change_target_fmt(get_default_data_format(l));
```

Comment:
`change_target_fmt` is only used in here and probably it could be replaced below.
`
if (use_default_format) { return MemoryDescriptorBuilder(l, get_default_data_format(l)).build();}
`

Reply:
Changed